packages/relay/src/lib/config/preconfiguredSpendingPlans.ts

```diff
@@ -0,0 +1,109 @@
+ * @returns {SpendingPlanConfig[]} An array of spending plan configurations.
+ * @throws {Error} If the configuration file is not found or cannot be read.
+ */
+const loadSpendingPlansConfig = (): SpendingPlanConfig[] => {
+  const path = findConfig(SPENDING_PLANS_CONFIG_FILE);
+  if (!path || !fs.existsSync(path)) {
+    throw new Error(`Configuration file not found at path: ${path || SPENDING_PLANS_CONFIG_FILE}`);
+  }
+  const rawData = fs.readFileSync(path, 'utf-8');
+  return JSON.parse(rawData) as SpendingPlanConfig[];
+};
+
+/**
+ * Populates the database with pre-configured spending plans.
+ * @param {Logger} logger - The logger instance.
+ * @param {HbarSpendingPlanRepository} hbarSpendingPlanRepository - The HBAR spending plan repository.
+ * @param {EthAddressHbarSpendingPlanRepository} ethAddressHbarSpendingPlanRepository - The ETH address HBAR spending plan repository.
+ * @param {IPAddressHbarSpendingPlanRepository} ipAddressHbarSpendingPlanRepository - The IP address HBAR spending plan repository.
+ * @returns {Promise<void>} - A promise that resolves when the operation is complete.
+ * @throws {Error} - If the spending plans configuration file is not found or cannot be loaded.
+ * @async
+ */
+export const populatePreconfiguredSpendingPlans = async (
+  logger: Logger,
+  hbarSpendingPlanRepository: HbarSpendingPlanRepository,
+  ethAddressHbarSpendingPlanRepository: EthAddressHbarSpendingPlanRepository,
+  ipAddressHbarSpendingPlanRepository: IPAddressHbarSpendingPlanRepository,
+): Promise<void> => {
+  const requestDetails = new RequestDetails({ requestId: '', ipAddress: '' });
+  const ttl = PRE_CONFIGURED_SPENDING_PLANS_TTL;
+
+  const spendingPlans = loadSpendingPlansConfig();
+
+  for (const plan of spendingPlans) {
+    const { name, ethAddresses, ipAddresses, subscriptionTier } = plan;
+
+    if (!ethAddresses?.length && !ipAddresses?.length) {
+      logger.warn(
+        `${requestDetails.formattedRequestId} Skipping HBAR spending plan '${name}' as it has no ethAddresses or ipAddresses`,
+      );
+      continue;
+    }
+
+    const spendingPlan = await hbarSpendingPlanRepository.create(subscriptionTier, requestDetails, ttl);
+    logger.info(`${requestDetails.formattedRequestId} Created HBAR spending plan '${name}' with ID ${spendingPlan.id}`);
+
+    if (ethAddresses) {
+      for (const ethAddress of ethAddresses) {
+        if (!(await ethAddressHbarSpendingPlanRepository.existsByAddress(ethAddress, requestDetails))) {
+          await ethAddressHbarSpendingPlanRepository.save({ ethAddress, planId: spendingPlan.id }, requestDetails, ttl);
+          logger.info(
+            `${requestDetails.formattedRequestId} Associated HBAR spending plan '${name}' with ETH address ${ethAddress}`,
+          );
+        } else {
+          logger.warn(
+            `${requestDetails.formattedRequestId} Skipping ETH address ${ethAddress} as it already has an HBAR spending plan associated with it`,
+          );
```

Comment:
Nit: Is this a warning or useful "info"?

Reply:
It's a warning, because if we get here, then the JSON file is most likely wrongly configured.

Each unique ETH address should be linked only to one plan, we cannot have it linked to two plans because we wouldn't know which plan to use in this case.

Reply:
Sure, changed it to a `trace` log.